test_scripts/API/Restructuring_OnResetTimeout/common_OnResetTimeout.lua

```diff
@@ -0,0 +1,559 @@
+    SetInteriorVehicleData = {
+      hmi_levels = { "FULL", "BACKGROUND", "LIMITED", "NONE" }
+    },
+    SendLocation = {
+      hmi_levels = { "FULL", "BACKGROUND", "LIMITED", "NONE" }
+    },
+    DialNumber = {
+      hmi_levels = { "FULL", "BACKGROUND", "LIMITED", "NONE" }
+    },
+    DiagnosticMessage = {
+      hmi_levels = { "FULL", "BACKGROUND", "LIMITED", "NONE" }
+    },
+    ButtonPress = {
+      hmi_levels = { "FULL", "BACKGROUND", "LIMITED", "NONE" }
+    }
+  }
+  for i = 1, pCountOfRCApps do
+    local appId = config["application" .. i].registerAppInterfaceParams.fullAppID
+    preloadedTable.policy_table.app_policies[appId] = c.getAppConfig(preloadedTable)
+    preloadedTable.policy_table.app_policies[appId].AppHMIType = nil
+  end
+  c.tableToJsonFile(preloadedTable, preloadedFile)
+end
+
+function c.preconditions(isPreloadedUpdate, pCountOfRCApps)
+  if isPreloadedUpdate == nil then isPreloadedUpdate = true end
+  if isPreloadedUpdate == true then
+    backupPreloadedPT()
+    updatePreloadedPT(pCountOfRCApps)
+  end
+end
+
+local function restorePreloadedPT()
+  local preloadedFile = commonFunctions:read_parameter_from_smart_device_link_ini("PreloadedPT")
+  commonPreconditions:RestoreFile(preloadedFile)
+end
+
+--[[ @onResetTimeoutNotification: Notification from HMI  in case the results long processing on HMI
+--! @parameters:
+--! pRequestID - Id between HMI and SDL which SDL used to send the request for method in question, for which timeout needs to be reset.
+--! pMethodName - Name of the function for which timeout needs to be reset.
+--! pResetPeriod - Timeout period in milliseconds, for the method for which timeout needs to be reset.
+--! @return: none
+--]]
+function c.onResetTimeoutNotification(pRequestID, pMethodName, pResetPeriod)
+  c.getHMIConnection():SendNotification("BasicCommunication.OnResetTimeout",
+    { requestID = pRequestID,
+    methodName = pMethodName,
+    resetPeriod = pResetPeriod
+  })
+  c.notificationTime = timestamp()
+end
+
+--[[ @responseWithOnResetTimeout: parameters from HMI for response
+--! @parameters:
+--! pData - Id between HMI and SDL which SDL used to send the request for method in question, for which timeout needs to be reset.
+--! pParams - parameters for BasicCommunication.OnResetTimeout
+--! @return: none
+--]]
+function c.responseWithOnResetTimeout(pData, pParams)
+  local function sendOnResetTimeout()
+    c.onResetTimeoutNotification(pData.id, pData.method, pParams.resetPeriod)
+  end
+  local function sendresponse()
+    c.getHMIConnection():SendResponse(pData.id, pData.method, "SUCCESS", {})
+  end
+  RUN_AFTER(sendresponse, pParams.respTime)
+  RUN_AFTER(sendOnResetTimeout, pParams.notificationTime)
+end
+
+--[[ @SendLocation: Successful processing SendLocation RPC
+--! @parameters:
+--! pRespExpect - The time during which the expected response to the mobile
+--! pRespTime - HMI response time after receiving the request
```

Comment:
pRespTime - Time between response on mobile app and sending the OnResetTimeout notification from HMI